pkg/config/config.go

```diff
@@ -261,3 +261,18 @@ func GetOktetoContextsStorePath() string {
 func GetCertificatePath() string {
 	return filepath.Join(GetOktetoHome(), ".ca.crt")
 }
+
+// GetDeployOrigin gets the pipeline deploy origin. This is the initiator of the
+// deploy action: web, cli, github-action, etc
+func GetDeployOrigin() (src string) {
+	src = os.Getenv(model.OktetoOriginEnvVar)
+	if src == "" {
+		src = "cli"
+	}
+	// deploys within another okteto deploy take precedence as a deploy origin.
+	// This is running okteto pipeline deploy as a step of another okteto deploy
+	if os.Getenv(model.OktetoWithinDeployCommandContextEnvVar) == "true" {
```

Comment:
Maybe we need to revisit the places where we are checking this env var. Deploy command sets it as false almost at the beginning of the command: https://github.com/okteto/okteto/blob/master/cmd/deploy/deploy.go#L131

I see several checks that I'm not sure they are working as expected

Reply:
This works for multi repo with `okteto pipeline deploy`. Eg: https://github.com/okteto/movies-api/blob/main/okteto-pipeline.yml#L3 

Not sure if `okteto deploy` inside another `okteto deploy`  is a thing? If so the metrics will be somewhat skewed but not sure how many of these occurrences there will be. I can always create another envvar

Reply:
I don't think it will be frequent. For now I wouldn't consider that case until users start using it.

There is one case that I'm not sure which should be the right value for the metric. If you execute `okteto deploy -b master` or `okteto deploy -r https://github.com/okteto/movies`, it will execute directly a pipeline https://github.com/okteto/okteto/blob/master/cmd/deploy/deploy.go#L116 but it wouldn't be considered within an okteto deploy command. Which honestly, not sure if it is right or not.

Also, bear in mind that a pipeline is executed in the backend as an `okteto deploy`, in case it affects somehow the metrics

Reply:
@ifbyol can you take another look? I added `OKTETO_SKIP_CONFIG_CREDENTIALS_UPDATE` for the proxy case you describe and kept the origin `OKTETO_WITHIN_DEPLOY_COMMAND_CONTEXT` independent from all that so we can use it safely everywhere else knowing it will not change

Reply:
@maroshii looks good to me, but there are other places where that variable was using before and not sure if they should use the new one or the old one. Looking in to the code, I would say they have to use `OKTETO_WITHIN_DEPLOY_COMMAND_CONTEXT`, but @jLopezbarb can you double check, please? If you have doubts about what we are talking about, let me know and we can do a quick call